aten/src/ATen/native/cuda/SortingKthValue.cu

```diff
@@ -0,0 +1,249 @@
+#include <ATen/ATen.h>
+#include <ATen/native/SortingUtils.h>
+#include <assert.h>
+#include <c10/macros/Macros.h>
+#include <stdlib.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/detail/TensorInfo.cuh>
+#include <THC/THCDeviceUtils.cuh> // only for THCRoundUp?
+#include <THC/THCNumerics.cuh>
+#include <THC/THCScanUtils.cuh>
+#include <THC/THCTensorMathReduce.cuh> // AddOp
+
+#include <thrust/device_ptr.h>
+#include <thrust/sort.h>
+
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/extrema.h>
+#include <thrust/inner_product.h>
+#include <thrust/sequence.h>
+#include <THC/THCThrustAllocator.cuh>
+#include <ATen/native/cuda/SortingCommon.cuh>
+#include <ATen/native/cuda/SortingRadixSelect.cuh>
+
+namespace at {
+namespace native {
+
+namespace {
+
+
+template <typename scalar_t, typename index_t, int Dim>
+__global__ void gatherKthValue(
+    cuda::detail::TensorInfo<scalar_t, index_t> input,
+    index_t inputSliceSize,
+    index_t k,
+
+    index_t numInputSlices,
+    index_t inputWithinSliceStride,
+
+    cuda::detail::TensorInfo<scalar_t, index_t> kthValue,
+    cuda::detail::TensorInfo<int64_t, index_t> indices) {
+  // Indices are limited to integer fp precision, so counts can fit in
+  // int32, regardless of index_t
+  __shared__ int smem[WARP_SIZE]; // one per each warp, up to warp limit
+
+  index_t slice = getLinearBlockId<index_t>();
+  if (slice >= numInputSlices) {
+    return;
+  }
```

Comment:
I see you are adding new CUDA code but you are not adding any tests on top of that. The [PR](https://github.com/pytorch/pytorch/pull/8406), that you are referencing in the description, is making changes in  test/data/test_cuda_ignores.txt in order to test new code, can you do the same, please?

Reply:
Thanks for catching this. I meant to include this, but didn't add it to the commit.